Sources/Swift/Integrations/Performance/IO/Data+SentryTracing.swift

```diff
@@ -0,0 +1,55 @@
+@_implementationOnly import _SentryPrivate
+
+/// A `Data` extension that automatically tracks read and write operations with Sentry.
+///
+/// - Note: Methods provided by this extension reflect the same functionality as the original `Data` methods,
+///         but they automatically track the operation with Sentry.
+public extension Data {
+
+    /// Initialize a `Data` with the contents of a `URL`, automatically tracking the operation with Sentry.
+    ///
+    /// - parameter url: The `URL` to read.
+    /// - parameter options: Options for the read operation. Default value is `[]`.
+    /// - throws: An error in the Cocoa domain, if `url` cannot be read.
+    init(contentsOfUrlWithSentryTracing url: URL, options: Data.ReadingOptions = []) throws {
+        let tracker = SentryFileIOTracker.sharedInstance()
+        // Using the bridging of `Data` to `NSData` caused issues on older versions of macOS.
+        // Therefore we do not use the `measureNSData` method.
+        if #available(iOS 18, macOS 15, tvOS 18, *) {
```

Comment:
`h`: I'm unsure about this approach. It still could occur that some OS versions switch back to bridging NSData, and then users would get duplicated spans. This also has the downside that people can't use this on older iOS versions when they, for example, disable swizzling. Instead, it would be great if we could implement some deduplication logic for duplicated spans. I still need to figure out how we can do this properly, and will get back to you.

Reply:
We agreed in Slack to skip a deduplication logic because it's complicated to get it right, it's a lot of work, and most people won't need it. Instead, we should allow the use of manual API before iOS 18. If users want file IO spans, they need to use the manual API anyway, and they can turn off auto file IO instrumentation in general. With this, they avoid having duplicates before iOS 18. We need to mention that people should consider turning off auto file IO instrumentation when using the manual API.